src/integer/mat_z/sample/discrete_gauss.rs

```diff
@@ -306,3 +368,88 @@ mod test_sample_d {
         let _ = MatZ::sample_d_common(10, 1024, 1.25f32).unwrap();
     }
 }
+
+#[cfg(test)]
+mod test_sample_d_common_non_spherical {
+    use crate::{
+        integer::{MatZ, Z},
+        rational::{MatQ, Q},
+        traits::{GetNumRows, Pow},
+    };
+    use std::str::FromStr;
+
+    /// Checks whether `sample_d_common_non_spherical` is available for all types
+    /// implementing [`Into<Z>`], i.e. u8, u16, u32, u64, i8, ...
+    /// or [`Into<Q>`], i.e. u8, i16, f32, Z, Q, ...
+    /// or [`Into<MatQ>`], i.e. MatQ, MatZ
+    #[test]
+    fn availability() {
+        let r = Q::from(8);
+        let convolution_matrix = MatQ::from_str("[[100,1],[1,65]]").unwrap();
+        let convolution_matrix = (convolution_matrix - r.pow(2).unwrap() * MatQ::identity(2, 2))
+            .cholesky_decomposition();
+
+        let _ = MatZ::sample_d_common_non_spherical(16, &convolution_matrix, 8).unwrap();
+
+        let _ = MatZ::sample_d_common_non_spherical(16u16, &convolution_matrix, 8_u16).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16u32, &convolution_matrix, 8_u32).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16u64, &convolution_matrix, 8_u64).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16i8, &convolution_matrix, 8_i8).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16i16, &convolution_matrix, 8_i16).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16i32, &convolution_matrix, 8_i32).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16i64, &convolution_matrix, 8_i64).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(Z::from(16), &convolution_matrix, Q::from(8))
+            .unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16, &convolution_matrix, Z::from(8)).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16, &convolution_matrix, 8f32).unwrap();
+        let _ = MatZ::sample_d_common_non_spherical(16, &convolution_matrix, 8f64).unwrap();
+    }
+
+    /// Checks whether the function panics if a non positive-definite matrix is provided.
+    #[test]
+    #[should_panic]
+    fn no_convolution_matrix_1() {
+        let r = Q::from(4);
+        let convolution_matrix = MatQ::from_str("[[-1,1],[1,1]]").unwrap();
+        let convolution_matrix = (convolution_matrix - r.pow(2).unwrap() * MatQ::identity(2, 2))
+            .cholesky_decomposition();
+
+        let _ = MatZ::sample_d_common_non_spherical(16, &convolution_matrix, r).unwrap();
+    }
```

Comment:
Please, make this an error too and return it, if it happens. Furthermore, it needs to be added to the doc-comment.